engine/src/main/java/org/camunda/bpm/engine/impl/cmmn/handler/TimerEventListenerItemHandler.java

```diff
@@ -43,6 +47,12 @@ protected void initializeActivity(CmmnElement element, CmmnActivity activity, Cm
   private void initializeTimerEventListenerJobDeclaration(CmmnElement element, CmmnActivity activity, CmmnHandlerContext context) {
     TimerEventListenerJobDeclaration timerEventListenerJobDeclaration = parseTimerExpression(element,context);
     activity.setProperty(ItemHandler.PROPERTY_TIMERVEVENTLISTENER_JOBDECLARATION, timerEventListenerJobDeclaration);
+
+    CaseDefinitionEntity caseDefinition = (CaseDefinitionEntity) context.getCaseDefinition();
```

Comment:
Instead of caching the declaration at the definition, we could use the given `CmmnHandlerContext` to cache them, so that it would look like that:

```
  private void initializeTimerEventListenerJobDeclaration(CmmnElement element, CmmnActivity activity, CmmnHandlerContext context) {
    TimerEventListenerJobDeclaration timerEventListenerJobDeclaration = parseTimerExpression(element,context);
    activity.setProperty(ItemHandler.PROPERTY_TIMERVEVENTLISTENER_JOBDECLARATION, timerEventListenerJobDeclaration);
    context.addJobDeclaration(timerEventListenerJobDeclaration);
  }
```